tools/clang/tools/dxil2spv/lib/dxil2spv.cpp

```diff
@@ -175,29 +175,41 @@ void Translator::createStageIOVariables(
   // Translate DXIL input signature to SPIR-V stage input vars.
   for (const std::unique_ptr<hlsl::DxilSignatureElement> &elem :
        inputSignature) {
-    clang::spirv::SpirvVariable *var = spvBuilder.addStageIOVar(
-        toSpirvType(elem.get()), spv::StorageClass::Input,
-        elem->GetSemanticName(), false, {});
-    unsigned id = elem->GetID();
-    interfaceVars.push_back(var);
-    inputSignatureElementMap[id] = var;
-
-    // Use unique DXIL signature element ID as SPIR-V Location.
-    spvBuilder.decorateLocation(var, id);
+    createStageIOVariable(elem.get());
   }
 
   // Translate DXIL input signature to SPIR-V stage ouput vars.
   for (const std::unique_ptr<hlsl::DxilSignatureElement> &elem :
        outputSignature) {
-    clang::spirv::SpirvVariable *var = spvBuilder.addStageIOVar(
-        toSpirvType(elem.get()), spv::StorageClass::Output,
-        elem->GetSemanticName(), false, {});
-    unsigned id = elem->GetID();
-    interfaceVars.push_back(var);
-    outputSignatureElementMap[id] = var;
+    createStageIOVariable(elem.get());
+  }
+}
+
+void Translator::createStageIOVariable(hlsl::DxilSignatureElement *elem) {
+  const spirv::SpirvType *spirvType = toSpirvType(elem);
```

Comment:
Could this be a `nullptr`? I can see that `toSpirvType` emits a diagnostic, but I'm not sure if that's fatal or not.

Reply:
It can be nullptr and the error emitting is not fatal. Since this is a problem throughout already that hasn't really been handled well (while I was debating introducing exceptions) I'll file a follow-up bug to address it uniformly.

Reply:
SG